### ISSUE TITLE
convert - Out of Memory (non-valid syntax)

### ISSUE THREAD

**Issue by cawalch** [asciidoctor/asciidoctor.js#760]
```javascript
const asciidoctor = require('asciidoctor')()
asciidoctor.convert(Buffer.from('103a3a0a393e0901', 'hex'))
```
**Version** 2.0.3

Found with [jsfuzz](https://github.com/fuzzitdev/jsfuzz)

**Comment by mojavelinux** [comment]
This causes Asciidoctor to hang as well. Here's the UTF-8 string to reproduce:

```
a::\n0>\tb
```

Looks like a dlist parsing error related to https://github.com/asciidoctor/asciidoctor/issues/2888.

**Comment by mojavelinux** [comment]
Yep, same exact problem. We made a change to the syntax rules, but forgot to update one of the matchers.
